From e714497ce829cb432bb9134005f4769fcb3f2176 Mon Sep 17 00:00:00 2001
From: Andreas Beckmann <anbe@debian.org>
Date: Sun, 19 Sep 2021 15:24:53 +0200
Subject: [PATCH 2/2] backport set_current_state changes from 470.63.01

---
 common/inc/nv-time.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/common/inc/nv-time.h b/common/inc/nv-time.h
index ee14004..0133383 100644
--- a/common/inc/nv-time.h
+++ b/common/inc/nv-time.h
@@ -214,7 +214,7 @@ static inline NV_STATUS nv_sleep_ms(unsigned int ms)
         // the requested timeout has expired, loop until less
         // than a jiffie of the desired delay remains.
         //
-        current->state = TASK_INTERRUPTIBLE;
+        set_current_state(TASK_INTERRUPTIBLE);
         do
         {
             schedule_timeout(jiffies);
-- 
2.20.1

