From e124caa6bc7adb6bc3eb7068b3643626118fe8e7 Mon Sep 17 00:00:00 2001
From: Andreas Beckmann <anbe@debian.org>
Date: Tue, 15 Dec 2020 17:09:29 +0100
Subject: [PATCH 2/2] backport get_dma_ops changes from 455.45.01

---
 common/inc/nv-linux.h | 4 ++++
 conftest.sh           | 9 +++++++++
 2 files changed, 13 insertions(+)

diff --git a/common/inc/nv-linux.h b/common/inc/nv-linux.h
index fb6597a..ccaad23 100644
--- a/common/inc/nv-linux.h
+++ b/common/inc/nv-linux.h
@@ -167,6 +167,10 @@ static inline uid_t __kuid_val(uid_t uid)
 #include <linux/pagemap.h>
 #include <linux/dma-mapping.h>
 
+#if defined(NV_LINUX_DMA_MAP_OPS_H_PRESENT)
+#include <linux/dma-map-ops.h>
+#endif
+
 #if defined(CONFIG_SWIOTLB) && defined(NVCPU_AARCH64)
 #include <linux/swiotlb.h>
 #endif
diff --git a/conftest.sh b/conftest.sh
index 4080d15..5d9c835 100755
--- a/conftest.sh
+++ b/conftest.sh
@@ -152,6 +152,7 @@ test_headers() {
     FILES="$FILES asm/opal-api.h"
     FILES="$FILES sound/hdaudio.h"
     FILES="$FILES asm/pgtable_types.h"
+    FILES="$FILES linux/dma-map-ops.h"
 
     translate_and_preprocess_header_files $FILES
 }
@@ -1536,8 +1537,16 @@ compile_test() {
             # e1c7e324539a ("dma-mapping: always provide the dma_map_ops
             # based implementation") in v4.5
             #
+            # Commit 0a0f0d8be76d ("dma-mapping: split <linux/dma-mapping.h>")
+            # in v5.10-rc1 (2020-09-22), moved get_dma_ops() function
+            # prototype from <linux/dma-mapping.h> to <linux/dma-map-ops.h>.
+            #
             CODE="
+            #if defined(NV_LINUX_DMA_MAP_OPS_H_PRESENT)
+            #include <linux/dma-map-ops.h>
+            #else
             #include <linux/dma-mapping.h>
+            #endif
             void conftest_get_dma_ops(void) {
                 get_dma_ops();
             }"
-- 
2.20.1

