From 80f0d7c4d9a82e7d73efd1669ffd631c4b7110c3 Mon Sep 17 00:00:00 2001
From: Andreas Beckmann <anbe@debian.org>
Date: Thu, 14 Apr 2022 22:34:31 +0200
Subject: [PATCH] backport set_{memory,pages}{,_array}_uc changes from
 510.39.01

---
 conftest.sh | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/conftest.sh b/conftest.sh
index e831b64..471d55c 100755
--- a/conftest.sh
+++ b/conftest.sh
@@ -466,6 +466,7 @@ compile_test() {
             # It does not exist on all architectures.
             #
             CODE="
+            #include <linux/types.h>
             #if defined(NV_ASM_SET_MEMORY_H_PRESENT)
             #if defined(NV_ASM_PGTABLE_TYPES_H_PRESENT)
             #include <asm/pgtable_types.h>
@@ -487,6 +488,7 @@ compile_test() {
             # It does not exist on all architectures.
             #
             CODE="
+            #include <linux/types.h>
             #if defined(NV_ASM_SET_MEMORY_H_PRESENT)
             #if defined(NV_ASM_PGTABLE_TYPES_H_PRESENT)
             #include <asm/pgtable_types.h>
@@ -543,6 +545,7 @@ compile_test() {
             # It does not exist on all architectures.
             #
             CODE="
+            #include <linux/types.h>
             #if defined(NV_ASM_SET_MEMORY_H_PRESENT)
             #if defined(NV_ASM_PGTABLE_TYPES_H_PRESENT)
             #include <asm/pgtable_types.h>
@@ -569,6 +572,7 @@ compile_test() {
             # 19 14:51:15 2009)
             #
             CODE="
+            #include <linux/types.h>
             #if defined(NV_ASM_SET_MEMORY_H_PRESENT)
             #if defined(NV_ASM_PGTABLE_TYPES_H_PRESENT)
             #include <asm/pgtable_types.h>
-- 
2.20.1

