From 3d584a19fce5245197d64f3f17d9b5994af6cb14 Mon Sep 17 00:00:00 2001
From: Andreas Beckmann <anbe@debian.org>
Date: Sat, 29 Aug 2020 00:23:41 +0200
Subject: [PATCH 19/34] work around mmap_{sem=>lock} rename

---
 Makefile    |  1 +
 conftest.sh | 18 ++++++++++++++++++
 nv-linux.h  |  4 ++++
 3 files changed, 23 insertions(+)

diff --git a/Makefile b/Makefile
index a022ffa..c3435c6 100644
--- a/Makefile
+++ b/Makefile
@@ -158,6 +158,7 @@ COMPILE_TESTS = \
 	proc_ops \
 	timeval \
 	vmalloc_has_pgprot_t_arg \
+	mm_has_mmap_lock \
 
 #
 # CFLAGS dependent on the type of builds (e.g. single/muliple module, debug)
diff --git a/conftest.sh b/conftest.sh
index 1eb8008..7247a19 100755
--- a/conftest.sh
+++ b/conftest.sh
@@ -2314,6 +2314,24 @@ compile_test() {
 
         ;;
 
+        mm_has_mmap_lock)
+            #
+            # Determine if the 'mm_struct' structure has a 'mmap_lock' field.
+            #
+            # Kernel commit da1c55f1b272 ("mmap locking API: rename mmap_sem
+            # to mmap_lock") replaced the field 'mmap_sem' by 'mmap_lock'
+            # in v5.8-rc1 (2020-06-08).
+            CODE="
+            #include <linux/mm_types.h>
+
+            int conftest_mm_has_mmap_lock(void) {
+                return offsetof(struct mm_struct, mmap_lock);
+            }"
+
+            compile_check_conftest "$CODE" "NV_MM_HAS_MMAP_LOCK" "" "types"
+
+        ;;
+
         # When adding a new conftest entry, please use the correct format for
         # specifying the relevant upstream Linux kernel commit.
         #
diff --git a/nv-linux.h b/nv-linux.h
index 050b154..120ff0c 100644
--- a/nv-linux.h
+++ b/nv-linux.h
@@ -2103,6 +2103,10 @@ static inline NvU64 nv_node_end_pfn(int nid)
 #endif
 }
 
+#if defined(NV_MM_HAS_MMAP_LOCK)
+#define mmap_sem mmap_lock
+#endif
+
 /* get_user_pages
  *
  * The 8-argument version of get_user_pages was deprecated by commit 
-- 
2.20.1

