From 1d8cdaf8ff6f3a3ab1579510c715ceaafa9b54a0 Mon Sep 17 00:00:00 2001
From: Andreas Beckmann <anbe@debian.org>
Date: Tue, 21 Dec 2021 12:43:03 +0100
Subject: [PATCH 27/34] add static and nv_ prefix to copied drm legacy bits

---
 conftest.sh |  1 +
 nv-drm.c    | 42 ++++++++++++++++++++++++++----------------
 2 files changed, 27 insertions(+), 16 deletions(-)

diff --git a/conftest.sh b/conftest.sh
index 264bb2a..b65ede5 100755
--- a/conftest.sh
+++ b/conftest.sh
@@ -156,6 +156,7 @@ test_headers() {
     FILES="$FILES drm/drm_ioctl.h"
     FILES="$FILES drm/drm_pci.h"
     FILES="$FILES drm/drm_legacy.h"
+    FILES="$FILES drm/drm_agpsupport.h"
     FILES="$FILES generated/autoconf.h"
     FILES="$FILES generated/compile.h"
     FILES="$FILES generated/utsrelease.h"
diff --git a/nv-drm.c b/nv-drm.c
index 2b4a414..4853498 100644
--- a/nv-drm.c
+++ b/nv-drm.c
@@ -126,6 +126,10 @@
 #include <drm/drm_legacy.h>
 #endif
 
+#if defined(NV_DRM_DRM_AGPSUPPORT_H_PRESENT)
+#include <drm/drm_agpsupport.h>
+#endif
+
 #if IS_ENABLED(CONFIG_DRM_LEGACY) && defined(NV_DRM_LEGACY_PCI_INIT_PRESENT)
 #define nv_drm_pci_init drm_legacy_pci_init
 #define nv_drm_pci_exit drm_legacy_pci_exit
@@ -135,7 +139,7 @@
 #else
 
 // copied from src:linux (5.9.11-1) drivers/gpu/drm/drm_legacy.h
-struct drm_agp_mem {
+struct nv_drm_agp_mem {
 	unsigned long handle;
 	struct agp_memory *memory;
 	unsigned long bound;
@@ -143,22 +147,24 @@ struct drm_agp_mem {
 	struct list_head head;
 };
 
+static inline
 // copied from src:linux (5.9.11-1) drivers/gpu/drm/drm_memory.c
 /** Wrapper around agp_free_memory() */
-void drm_free_agp(struct agp_memory *handle, int pages)
+void nv_drm_free_agp(struct agp_memory *handle, int pages)
 {
 	agp_free_memory(handle);
 }
 
+static inline
 // copied from src:linux (5.9.11-1) drivers/gpu/drm/drm_memory.c
 /** Wrapper around agp_unbind_memory() */
-int drm_unbind_agp(struct agp_memory *handle)
+int nv_drm_unbind_agp(struct agp_memory *handle)
 {
 	return agp_unbind_memory(handle);
 }
 
 // copied from src:linux (5.9.11-1) drivers/gpu/drm/drm_pci.c
-static void drm_pci_agp_init(struct drm_device *dev)
+static void nv_drm_pci_agp_init(struct drm_device *dev)
 {
 	if (drm_core_check_feature(dev, DRIVER_USE_AGP)) {
 		if (pci_find_capability(dev->pdev, PCI_CAP_ID_AGP))
@@ -172,6 +178,7 @@ static void drm_pci_agp_init(struct drm_device *dev)
 	}
 }
 
+static
 // copied from src:linux (5.9.11-1) drivers/gpu/drm/drm_agpsupport.c
 /**
  * drm_legacy_agp_clear - Clear AGP resource list
@@ -183,9 +190,9 @@ static void drm_pci_agp_init(struct drm_device *dev)
  *
  * Cleanup is only done for drivers who have DRIVER_LEGACY set.
  */
-void drm_legacy_agp_clear(struct drm_device *dev)
+void nv_drm_legacy_agp_clear(struct drm_device *dev)
 {
-	struct drm_agp_mem *entry, *tempe;
+	struct nv_drm_agp_mem *entry, *tempe;
 
 	if (!dev->agp)
 		return;
@@ -194,8 +201,8 @@ void drm_legacy_agp_clear(struct drm_device *dev)
 
 	list_for_each_entry_safe(entry, tempe, &dev->agp->memory, head) {
 		if (entry->bound)
-			drm_unbind_agp(entry->memory);
-		drm_free_agp(entry->memory, entry->pages);
+			nv_drm_unbind_agp(entry->memory);
+		nv_drm_free_agp(entry->memory, entry->pages);
 		kfree(entry);
 	}
 	INIT_LIST_HEAD(&dev->agp->memory);
@@ -207,19 +214,20 @@ void drm_legacy_agp_clear(struct drm_device *dev)
 	dev->agp->enabled = 0;
 }
 
+static
 // copied from src:linux (5.9.11-1) drivers/gpu/drm/drm_pci.c
-void drm_pci_agp_destroy(struct drm_device *dev)
+void nv_drm_pci_agp_destroy(struct drm_device *dev)
 {
 	if (dev->agp) {
 		arch_phys_wc_del(dev->agp->agp_mtrr);
-		drm_legacy_agp_clear(dev);
+		nv_drm_legacy_agp_clear(dev);
 		kfree(dev->agp);
 		dev->agp = NULL;
 	}
 }
 
 // copied from src:linux (5.9.11-1) drivers/gpu/drm/drm_pci.c
-static int drm_get_pci_dev(struct pci_dev *pdev,
+static int nv_drm_get_pci_dev(struct pci_dev *pdev,
 			   const struct pci_device_id *ent,
 			   struct drm_driver *driver)
 {
@@ -244,7 +252,7 @@ static int drm_get_pci_dev(struct pci_dev *pdev,
 	if (drm_core_check_feature(dev, DRIVER_MODESET))
 		pci_set_drvdata(pdev, dev);
 
-	drm_pci_agp_init(dev);
+	nv_drm_pci_agp_init(dev);
 
 	ret = drm_dev_register(dev, ent->driver_data);
 	if (ret)
@@ -258,13 +266,14 @@ static int drm_get_pci_dev(struct pci_dev *pdev,
 	return 0;
 
 err_agp:
-	drm_pci_agp_destroy(dev);
+	nv_drm_pci_agp_destroy(dev);
 	pci_disable_device(pdev);
 err_free:
 	drm_dev_put(dev);
 	return ret;
 }
 
+static
 // copied from src:linux (5.9.11-1) drivers/gpu/drm/drm_pci.c
 /**
  * drm_legacy_pci_init - shadow-attach a legacy DRM PCI driver
@@ -275,7 +284,7 @@ err_free:
  *
  * Return: 0 on success or a negative error code on failure.
  */
-int drm_legacy_pci_init(struct drm_driver *driver, struct pci_driver *pdriver)
+int nv_drm_pci_init(struct drm_driver *driver, struct pci_driver *pdriver)
 {
 	struct pci_dev *pdev = NULL;
 	const struct pci_device_id *pid;
@@ -306,12 +315,13 @@ int drm_legacy_pci_init(struct drm_driver *driver, struct pci_driver *pdriver)
 
 			/* stealth mode requires a manual probe */
 			pci_dev_get(pdev);
-			drm_get_pci_dev(pdev, pid, driver);
+			nv_drm_get_pci_dev(pdev, pid, driver);
 		}
 	}
 	return 0;
 }
 
+static
 // copied from src:linux (5.9.11-1) drivers/gpu/drm/drm_pci.c
 /**
  * drm_legacy_pci_exit - unregister shadow-attach legacy DRM driver
@@ -321,7 +331,7 @@ int drm_legacy_pci_init(struct drm_driver *driver, struct pci_driver *pdriver)
  * Unregister a DRM driver shadow-attached through drm_legacy_pci_init(). This
  * is deprecated and only used by dri1 drivers.
  */
-void drm_legacy_pci_exit(struct drm_driver *driver, struct pci_driver *pdriver)
+void nv_drm_pci_exit(struct drm_driver *driver, struct pci_driver *pdriver)
 {
 	struct drm_device *dev, *tmp;
 
-- 
2.20.1

