From e7b3bc1ceac645d80a2e063b2e0bc06bf628629c Mon Sep 17 00:00:00 2001
From: Andreas Beckmann <anbe@debian.org>
Date: Sun, 25 Feb 2024 09:38:15 +0100
Subject: [PATCH] backport pfn_valid changes in nv-mmap.c from 470.239.06

---
 nvidia/nv-mmap.c | 7 +++----
 1 file changed, 3 insertions(+), 4 deletions(-)

diff --git a/nvidia/nv-mmap.c b/nvidia/nv-mmap.c
index 5926d3c..190cda5 100644
--- a/nvidia/nv-mmap.c
+++ b/nvidia/nv-mmap.c
@@ -12,6 +12,7 @@
 
 #include "os-interface.h"
 #include "nv-linux.h"
+#include "nv-ibmnpu.h"
 #include "nv_speculation_barrier.h"
 
 /*
@@ -570,11 +571,9 @@ int nvidia_mmap_helper(
             //
             // This path is similar to the sysmem mapping code.
             // TODO: Refactor is needed as part of bug#2001704.
-            // Use pfn_valid to determine whether the physical address has
-            // backing struct page. This is used to isolate P8 from P9.
             //
-            if (!IS_REG_OFFSET(nv, access_start, access_len) &&
-                (pfn_valid(PFN_DOWN(mmap_start))))
+            if ((nv_get_numa_status(nvl) == NV_NUMA_STATUS_ONLINE) &&
+                !IS_REG_OFFSET(nv, access_start, access_len))
             {
                 ret = nvidia_mmap_numa(vma, mmap_context);
                 if (ret)
-- 
2.20.1

