From 290f163beb619ddcffe5ec72d9bed15eb72faa59 Mon Sep 17 00:00:00 2001
From: Andreas Beckmann <anbe@debian.org>
Date: Wed, 19 Oct 2022 13:49:43 +0200
Subject: [PATCH] backport nv_install_notifier changes from 418.30

---
 nv-acpi.c | 44 +++++++++++++++++---------------------------
 1 file changed, 17 insertions(+), 27 deletions(-)

diff --git a/nv-acpi.c b/nv-acpi.c
index 7484a4d..ad29b77 100644
--- a/nv-acpi.c
+++ b/nv-acpi.c
@@ -476,20 +476,20 @@ static int nv_acpi_match(struct acpi_device *device, struct acpi_driver *driver)
 #endif
 
 /* Do the necessary allocations and install notifier "handler" on the device-node "device" */
-static NV_STATUS nv_install_notifier(struct acpi_device *device, acpi_notify_handler handler)
-{        
+static nv_acpi_t* nv_install_notifier(struct acpi_device *device, acpi_notify_handler handler)
+{
     nvidia_stack_t *sp = NULL;
     nv_acpi_t *pNvAcpiObject = NULL;
     NV_STATUS rmStatus = NV_ERR_GENERIC;
     acpi_status status = -1;
 
-    if ((!device) || device->driver_data)
-        return NV_ERR_GENERIC;
+    if (!device)
+        return NULL;
 
     NV_KMEM_CACHE_ALLOC_STACK(sp);
     if (sp == NULL)
     {
-        return NV_ERR_NO_MEMORY;
+        return NULL;
     }
 
     rmStatus = os_alloc_mem((void **) &pNvAcpiObject, sizeof(nv_acpi_t));
@@ -498,9 +498,6 @@ static NV_STATUS nv_install_notifier(struct acpi_device *device, acpi_notify_han
 
     os_mem_set((void *)pNvAcpiObject, 0, sizeof(nv_acpi_t));
 
-    // assign driver data structure ptr to this device
-    device->driver_data = pNvAcpiObject;
-
     // store a device reference in our object
     pNvAcpiObject->device = device;
     pNvAcpiObject->sp = sp;
@@ -510,33 +507,25 @@ static NV_STATUS nv_install_notifier(struct acpi_device *device, acpi_notify_han
     if (!ACPI_FAILURE(status)) 
     {
         pNvAcpiObject->notify_handler_installed = 1;
-        return NV_OK;
+        return pNvAcpiObject;
     }
 
-
 return_error:
-
      NV_KMEM_CACHE_FREE_STACK(sp);
      if (pNvAcpiObject)
          os_free_mem((void *)pNvAcpiObject);
-     device->driver_data = NULL;
-     return NV_ERR_GENERIC;
 
+     return NULL;
 }
 
 /* Tear-down and remove whatever nv_install_notifier did */ 
-static void nv_uninstall_notifier(struct acpi_device *device, acpi_notify_handler handler)
+static void nv_uninstall_notifier(nv_acpi_t *pNvAcpiObject, acpi_notify_handler handler)
 {
     acpi_status status;
-    nv_acpi_t *pNvAcpiObject = NULL;
 
-    if (!device)
-        return;
-
-    pNvAcpiObject = device->driver_data;
     if (pNvAcpiObject && pNvAcpiObject->notify_handler_installed)
     {
-        status = acpi_remove_notify_handler(device->handle, ACPI_DEVICE_NOTIFY, handler);
+        status = acpi_remove_notify_handler(pNvAcpiObject->device->handle, ACPI_DEVICE_NOTIFY, handler);
         if (ACPI_FAILURE(status))
         {
             nv_printf(NV_DBG_INFO,
@@ -546,14 +535,12 @@ static void nv_uninstall_notifier(struct acpi_device *device, acpi_notify_handle
         {
             NV_KMEM_CACHE_FREE_STACK(pNvAcpiObject->sp);
             os_free_mem((void *)pNvAcpiObject);
-            device->driver_data = NULL;
         }
     }
-   
+
     return;
 }
 
-
 /*
  * acpi methods init function.
  * check if the NVIF, _DSM and WMMX methods are present in the acpi namespace.
@@ -563,7 +550,6 @@ static void nv_uninstall_notifier(struct acpi_device *device, acpi_notify_handle
 void NV_API_CALL nv_acpi_methods_init(NvU32 *handlesPresent)
 {
     struct acpi_device *device = NULL;
-    NV_STATUS rmStatus;
     int retVal = -1;
 
 
@@ -596,8 +582,10 @@ void NV_API_CALL nv_acpi_methods_init(NvU32 *handlesPresent)
                            nodes' structures. So nothing more to be done */
             }
 
-            rmStatus = nv_install_notifier(device, nv_acpi_event);
-            if (rmStatus != NV_OK)
+            device->driver_data  = nv_install_notifier(device, nv_acpi_event);
+
+
+            if (!device->driver_data)
                 nvif_parent_gpu_handle = NULL;
 
         } while (0);
@@ -660,7 +648,9 @@ void NV_API_CALL nv_acpi_methods_uninit(void)
 
     acpi_bus_get_device(nvif_parent_gpu_handle, &device);
 
-    nv_uninstall_notifier(device, nv_acpi_event);
+    nv_uninstall_notifier(device->driver_data, nv_acpi_event);
+
+    device->driver_data = NULL;
     nvif_parent_gpu_handle = NULL;
 
     return;
-- 
2.20.1

