From a14a200951f8317907c10dda09d739506c235e3d Mon Sep 17 00:00:00 2001
From: Andreas Beckmann <anbe@debian.org>
Date: Sat, 6 Aug 2022 11:14:05 +0200
Subject: [PATCH] backport nv_install_notifier changes from 470.141.03

---
 nv-acpi.c  | 20 +++++++++-----------
 nv-linux.h |  1 +
 2 files changed, 10 insertions(+), 11 deletions(-)

diff --git a/nv-acpi.c b/nv-acpi.c
index 2135b4e..b8c9332 100644
--- a/nv-acpi.c
+++ b/nv-acpi.c
@@ -476,14 +476,14 @@ static int nv_acpi_match(struct acpi_device *device, struct acpi_driver *driver)
 #endif
 
 /* Do the necessary allocations and install notifier "handler" on the device-node "device" */
-static nv_acpi_t* nv_install_notifier(struct acpi_device *device, acpi_notify_handler handler)
+static nv_acpi_t* nv_install_notifier(struct acpi_handle *handle, acpi_notify_handler handler)
 {
     nvidia_stack_t *sp = NULL;
     nv_acpi_t *pNvAcpiObject = NULL;
     NV_STATUS rmStatus = NV_ERR_GENERIC;
     acpi_status status = -1;
 
-    if (!device)
+    if (!handle)
         return NULL;
 
     NV_KMEM_CACHE_ALLOC_STACK(sp);
@@ -498,11 +498,11 @@ static nv_acpi_t* nv_install_notifier(struct acpi_device *device, acpi_notify_ha
 
     os_mem_set((void *)pNvAcpiObject, 0, sizeof(nv_acpi_t));
 
-    // store a device reference in our object
-    pNvAcpiObject->device = device;
+    // store a handle reference in our object
+    pNvAcpiObject->handle = handle;
     pNvAcpiObject->sp = sp;
 
-    status = acpi_install_notify_handler(device->handle, ACPI_DEVICE_NOTIFY,
+    status = acpi_install_notify_handler(handle, ACPI_DEVICE_NOTIFY,
               handler, pNvAcpiObject);
     if (!ACPI_FAILURE(status)) 
     {
@@ -525,7 +525,7 @@ static void nv_uninstall_notifier(nv_acpi_t *pNvAcpiObject, acpi_notify_handler
 
     if (pNvAcpiObject && pNvAcpiObject->notify_handler_installed)
     {
-        status = acpi_remove_notify_handler(pNvAcpiObject->device->handle, ACPI_DEVICE_NOTIFY, handler);
+        status = acpi_remove_notify_handler(pNvAcpiObject->handle, ACPI_DEVICE_NOTIFY, handler);
         if (ACPI_FAILURE(status))
         {
             nv_printf(NV_DBG_INFO,
@@ -554,20 +554,18 @@ void NV_API_CALL nv_acpi_methods_init(NvU32 *handlesPresent)
     int retVal = -1;
 #endif
 
-
     if (!handlesPresent) // Caller passed us invalid pointer.
         return;
 
-
     *handlesPresent = 0;
 
     NV_ACPI_WALK_NAMESPACE(ACPI_TYPE_DEVICE, ACPI_ROOT_OBJECT,
                         ACPI_UINT32_MAX, nv_acpi_find_methods, NULL, NULL);
 
-#if defined(NV_ACPI_BUS_GET_DEVICE_PRESENT)
     if (nvif_handle)
     {
         *handlesPresent = NV_ACPI_NVIF_HANDLE_PRESENT;
+#if defined(NV_ACPI_BUS_GET_DEVICE_PRESENT)
         do
         {
             if (!nvif_parent_gpu_handle) /* unknown error */
@@ -585,15 +583,15 @@ void NV_API_CALL nv_acpi_methods_init(NvU32 *handlesPresent)
                            nodes' structures. So nothing more to be done */
             }
 
-            device->driver_data  = nv_install_notifier(device, nv_acpi_event);
+            device->driver_data  = nv_install_notifier(device->handle, nv_acpi_event);
 
 
             if (!device->driver_data)
                 nvif_parent_gpu_handle = NULL;
 
         } while (0);
-    }
 #endif
+    }
 
     if (wmmx_handle)
         *handlesPresent = *handlesPresent | NV_ACPI_WMMX_HANDLE_PRESENT;
diff --git a/nv-linux.h b/nv-linux.h
index 49c7973..7c3478a 100644
--- a/nv-linux.h
+++ b/nv-linux.h
@@ -1774,6 +1774,7 @@ typedef struct
 {
     nv_stack_t *sp;
     struct acpi_device *device;
+    struct acpi_handle *handle;
 
     nv_video_t pNvVideo[NV_MAXNUM_DISPLAY_DEVICES];
 
-- 
2.20.1

