From 5b840832514ab3025124493218d70bfa57470c9c Mon Sep 17 00:00:00 2001
From: Andreas Beckmann <anbe@debian.org>
Date: Wed, 31 Jan 2024 03:06:19 +0100
Subject: [PATCH] fix typos

---
 nvidia-uvm/uvm_perf_thrashing.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/nvidia-uvm/uvm_perf_thrashing.c b/nvidia-uvm/uvm_perf_thrashing.c
index a81d0fc..fdb0e68 100644
--- a/nvidia-uvm/uvm_perf_thrashing.c
+++ b/nvidia-uvm/uvm_perf_thrashing.c
@@ -318,7 +318,7 @@ module_param(uvm_perf_thrashing_max_resets,    uint, S_IRUGO);
 unsigned uvm_perf_map_remote_on_native_atomics_fault = 0;
 module_param(uvm_perf_map_remote_on_native_atomics_fault, uint, S_IRUGO);
 
-// Global post-processed values of the module parameters. They can be overriden
+// Global post-processed values of the module parameters. They can be overridden
 // per VA-space.
 static bool g_uvm_perf_thrashing_enable;
 static unsigned g_uvm_perf_thrashing_threshold;
@@ -566,7 +566,7 @@ static void va_space_thrashing_info_init_params(va_space_thrashing_info_t *va_sp
     va_space_thrashing->params.threshold     = g_uvm_perf_thrashing_threshold;
     va_space_thrashing->params.pin_threshold = g_uvm_perf_thrashing_pin_threshold;
 
-    // Default thrashing parameters are overriden for simulated/emulated GPUs
+    // Default thrashing parameters are overridden for simulated/emulated GPUs
     if (g_uvm_global.num_simulated_devices > 0 &&
         (g_uvm_perf_thrashing_lapse_usec == UVM_PERF_THRASHING_LAPSE_USEC_DEFAULT)) {
         va_space_thrashing->params.lapse_ns  = UVM_PERF_THRASHING_LAPSE_USEC_DEFAULT_EMULATION * 1000;
-- 
2.20.1

